server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class GetHealthAction extends ActionType<GetHealthAction.Response> {
+
+    public static final GetHealthAction INSTANCE = new GetHealthAction();
+    // TODO: Need new name maybe
+    // cluster:health/get
+    public static final String NAME = "cluster:monitor/health2";
```

Comment:
I would suggest to use `cluster:health` here;

Reply:
It ties to the security model we want for this new type of action. I also think we should avoid introducing another action prefix as we already have `cluster:admin`, `cluster:monitor` and `cluster:internal`.

Reply:
Make sense. Though I wonder if we want to expose it even to unauthenticated users or may be to someone with very limited permissions?

Reply:
Still open to discussion here.

Reply:
I think we should keep it at `monitor`, since it could expose quite sensitive data too. We also then avoid having to add roles at this stage of the project.

I do think it could be a good idea to maybe expose a much more summarized status for the cluster available to more users (like a banner at the top of kibana saying "we are experiencing problems, sorry for the inconvience") but I think that would require a more dedicated API (or to have security checks integrated into this API).

Reply:
It is currently `"cluster:monitor/health_api"`. I assume that is okay with what you're saying here?